tests/lib/wheel.py

```diff
@@ -85,13 +83,11 @@ def make_metadata_file(
     if value is not _default:
         return File(path, ensure_binary(value))
 
-    metadata = CaseInsensitiveDict(
-        {
-            "Metadata-Version": "2.1",
-            "Name": name,
-            "Version": version,
-        }
-    )
+    metadata: dict[str, HeaderValue] = {
```

Comment:
This is a behaviour change. Technically, I think you could have used `typing.Mapping` instead of `dict`, and kept the insensitive dict usage.

I don't have an opinion on whether this is a good change or not, but wonder if it makes sense to make a separate PR to have the merit of it assessed separately?